src/net/ossl.cc

```diff
@@ -828,7 +828,7 @@ class session : public enable_shared_from_this<session>, public session_impl {
     }
 
     future<> do_shutdown() {
-        if(_error || !connected() || eof()) {
+        if(_error || !connected()) {
```

Comment:
Can a test be added?

Reply:
There are a number of shutdown unit tests in Seastar that verify the shutdown behavior